nx-X11/programs/Xserver/randr/randr.c

```diff
@@ -423,36 +572,147 @@ TellChanged(WindowPtr pWin, void *value)
                 if (output->changed)
                     RRDeliverOutputEvent(client, pWin, output);
             }
+
+#ifndef NXAGENT_SERVER
+            xorg_list_for_each_entry(iter, &pScreen->output_slave_list, output_head) {
+                pSlaveScrPriv = rrGetScrPriv(iter);
+                for (i = 0; i < pSlaveScrPriv->numOutputs; i++) {
+                    RROutputPtr output = pSlaveScrPriv->outputs[i];
+
+                    if (output->changed)
+                        RRDeliverOutputEvent(client, pWin, output);
+                }
+            }
+#endif
+        }
+
+#ifndef NXAGENT_SERVER
+        if (pRREvent->mask & RRProviderChangeNotifyMask) {
+            xorg_list_for_each_entry(iter, &pScreen->output_slave_list, output_head) {
+                pSlaveScrPriv = rrGetScrPriv(iter);
+                if (pSlaveScrPriv->provider->changed)
+                    RRDeliverProviderEvent(client, pWin, pSlaveScrPriv->provider);
+            }
+            xorg_list_for_each_entry(iter, &pScreen->offload_slave_list, offload_head) {
+                pSlaveScrPriv = rrGetScrPriv(iter);
+                if (pSlaveScrPriv->provider->changed)
+                    RRDeliverProviderEvent(client, pWin, pSlaveScrPriv->provider);
+            }
+            xorg_list_for_each_entry(iter, &pScreen->unattached_list, unattached_head) {
+                pSlaveScrPriv = rrGetScrPriv(iter);
+                if (pSlaveScrPriv->provider->changed)
+                    RRDeliverProviderEvent(client, pWin, pSlaveScrPriv->provider);
+            }
+        }
+#endif
+
+        if (pRREvent->mask & RRResourceChangeNotifyMask) {
+            if (pScrPriv->resourcesChanged) {
+                RRDeliverResourceEvent(client, pWin);
+            }
         }
     }
     return WT_WALKCHILDREN;
 }
 
+void
+RRSetChanged(ScreenPtr pScreen)
+{
+    /* set changed bits on the master screen only */
+#ifndef NXAGENT_SERVER
+    ScreenPtr master;
+#endif
+
+    rrScrPriv(pScreen);
+    rrScrPrivPtr mastersp;
+
+#ifndef NXAGENT_SERVER
+    if (pScreen->isGPU) {
+        master = pScreen->current_master;
+        if (!master)
+            return;
+        mastersp = rrGetScrPriv(master);
+    }
+    else
+#endif
+    {
+#ifndef NXAGENT_SERVER
+        master = pScreen;
+#endif
+        mastersp = pScrPriv;
+    }
+
+    mastersp->changed = TRUE;
+}
+
```

Comment:
Could be simplified and replaced by two lines:

```
rrScrPriv(pScreen);
pScrPriv->changed = TRUE;
```
So one #ifndef would be enough.